### PR TITLE
Add docs for the diff.jl methods

### PR DESCRIPTION
**Pull request by kshyatt**
If you think it would be helpful, I can try to generate some examples for the two `diff_tree` methods.

### CODE REVIEW

File: base/libgit2/diff.jl [hunk 1]
```diff
@@ -11,7 +11,18 @@ function Base.unsafe_convert(::Type{Ptr{DiffOptionsStruct}}, rr::Tuple{Ref{DiffO
     Base.unsafe_convert(Ptr{DiffOptionsStruct}, first(rr))
 end
 
+"""
+    diff_tree(repo::GitRepo, tree::GitTree, pathspecs::AbstractString=""; cached::Bool=false)
 
+Generates a [`GitDiff`](@ref) between `tree` (which will be used for the "old"
```
Comment (ararslan) [inline base/libgit2/diff.jl]: "Generate" perhaps?

File: base/libgit2/diff.jl [hunk 2]
```diff
@@ -34,6 +55,13 @@ function diff_tree(repo::GitRepo, oldtree::GitTree, newtree::GitTree)
     return GitDiff(repo, diff_ptr_ptr[])
 end
 
+"""
+    GitDiffStats(diff::GitDiff)
+
+Get the diff statistics from the `GitDiff` `diff`. This object records a summary
```
Comment (ararslan) [inline base/libgit2/diff.jl]: Maybe `@ref` the `GitDiff`?

**Comment by kshyatt** [comment]
`make check-whitespace` passes

**Comment by tkelman** [comment]
should also be checking `make docs` works

**Comment by kshyatt** [comment]
`make docs` works

**Comment by kshyatt** [comment]
Bump - ok to go?

**Review by ararslan — APPROVED** [review]

**Pull request merged**